### PR TITLE
docs: [FC-0074] add more detailed concepts on events

### PR DESCRIPTION
**Pull request by mariajgrimaldi**
### Description
This PR improves the concept section of the documentation by adding a new document dedicated only to Open edX Events.

Initially, I also added two additional modifications:
1. Update references in the Hooks Extension Framework document and explain why the framework should be adopted.
2. Adding a new document comparing Open edX Events vs Filters to ease adoption.

Which I later dropped in favor of centralizing shared documentation docs.openedx.org and avoiding duplication across repos: So I'll be moving those docs to docs.openedx.org, Here's the PR for the implementation: https://github.com/openedx/docs.openedx.org/pull/599

For more details on the decision, please see [this](https://github.com/openedx/openedx-filters/pull/222#issuecomment-2454579752) thread.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @mariajgrimaldi!

### What's next?

*Please work through the following steps to get your changes ready for engineering review:*

#### :radio_button: Get product approval

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.

#### :radio_button: Provide context

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads





#### :radio_button: Get a green build

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.


#### :radio_button: Update the status of your PR

Your PR is currently marked as a draft.  After completing the steps above, update its status by clicking "Ready for Review", or removing "WIP" from the title, as appropriate.




#### :radio_button: Let us know that your PR is ready for review:

### Who will review my changes?


This repository is currently maintained by `@openedx/hooks-extension-framework`.  Tag them in a comment and let them know that your changes are ready for review.


### Where can I find more information?

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)

### When can I expect my changes to be merged?

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*

File: docs/concepts/openedx-events.rst
Comment (mariajgrimaldi) [inline docs/concepts/openedx-events.rst:11]: I think we need more reasons here explaining why Open edX Events are important besides what they allow us to do.

File: docs/concepts/openedx-events.rst
Comment (mariajgrimaldi) [inline docs/concepts/openedx-events.rst:19]: I think it'd be helpful to add a short list of how they're used in the community.

File: docs/concepts/events-vs-filters.rst
Comment (mariajgrimaldi) [inline docs/concepts/events-vs-filters.rst:19]: Here we should use the official definition for both, so we don't maintain definitions in multiple places.

File: docs/concepts/events-vs-filters.rst
Comment (mariajgrimaldi) [inline docs/concepts/events-vs-filters.rst:40]:
```suggestion
|  **Definition**    |  Defined using the ``OpenEdxPublicSignal`` class, which                  |  Defined using the ``OpenEdxPublicFilter`` class, which provides a  |
```

File: docs/concepts/events-vs-filters.rst
Comment (mariajgrimaldi) [inline docs/concepts/events-vs-filters.rst:70]: We can be more specific here as with the other examples.

**Comment by mariajgrimaldi** [comment]
@sarina: here's the WIP I've been working on for the _1.1 Improve Definitions and Update Outdated References in the Framework_ project feature.

File: docs/concepts/openedx-events.rst
Comment (mariajgrimaldi) [reply docs/concepts/openedx-events.rst]: I decided to move this to a more generic "why to use the Open edX Hooks" section: https://github.com/openedx/openedx-events/pull/406/files#diff-d186fdffb21d5d2deb62cdc3cc4135a6bebd1d2e22a4218cc6ad5c6647f6d9a0R25-R54

File: docs/concepts/openedx-events.rst
Comment (mariajgrimaldi) [reply docs/concepts/openedx-events.rst]: Yes, I'll be working on this.

File: docs/concepts/openedx-events.rst
Comment (sarina) [inline docs/concepts/openedx-events.rst:18]: Maybe have "Event Bus" link to documentation on the event bus?

File: docs/concepts/openedx-events.rst
Comment (sarina) [inline docs/concepts/openedx-events.rst:27]:
```suggestion
#. An Open edX application component emits an event by calling the ``send_event`` method implemented by ``OpenEdxPublicEvents``, which calls the Django signal ``send`` method under the hood.

```

Using `#.` will auto-make a numbered list in rST. You need to include a blank line after each bullet point.

File: docs/concepts/openedx-events.rst
Comment (sarina) [inline docs/concepts/openedx-events.rst:29]: Who is "we"?

File: docs/concepts/openedx-events.rst
Comment (sarina) [inline docs/concepts/openedx-events.rst:30]: What does it mean to register an event receiver? Maybe link to documentation on this?

File: docs/concepts/openedx-events.rst
Comment (sarina) [inline docs/concepts/openedx-events.rst:31]:
```suggestion
#. Each event receiver performs additional processing or triggers other actions based on the event data.

```

File: docs/concepts/openedx-events.rst
Comment (sarina) [inline docs/concepts/openedx-events.rst:36]:
```suggestion
#. A user enrolls in a course, `triggering the COURSE_ENROLLMENT_CREATED event`_. This event includes information about the user, course, and enrollment details.

```

File: docs/concepts/openedx-events.rst
Comment (sarina) [inline docs/concepts/openedx-events.rst:37]: This link is a little confusing, because the text is generic but you're referencing a specific handler. How about:

```suggestion
#. A signal receiver listening for the ``COURSE_ENROLLMENT_CREATED`` event is called and processes the event data. In this case, it would be the `course_enrollment_post_save receiver`_.

```

File: docs/concepts/openedx-events.rst
Comment (sarina) [inline docs/concepts/openedx-events.rst:34]:
```suggestion
Here is an example of an event that triggers email in response to course enrollment in action:
```

File: docs/concepts/openedx-events.rst
Comment (sarina) [inline docs/concepts/openedx-events.rst:39]: Is there code or something that "considers" the event complete? I guess I don't know exactly what this means.

File: docs/concepts/openedx-events.rst
Comment (sarina) [inline docs/concepts/openedx-events.rst:46]: I think it would be good to include some detail about how you don't need to modify code within the `openedx` github org to achieve this. Could we link out to some third-party examples?

**Review by sarina — CHANGES_REQUESTED** [review]

File: docs/concepts/openedx-events.rst
Comment (mariajgrimaldi) [reply docs/concepts/openedx-events.rst]: Done! But the only available docs we have to date about the event bus is a confluence document, so that's what I linked: https://openedx.atlassian.net/wiki/spaces/AC/pages/3508699151/How+to+start+using+the+Event+Bus

File: docs/concepts/openedx-events.rst
Comment (mariajgrimaldi) [reply docs/concepts/openedx-events.rst]: Thanks!

File: docs/concepts/openedx-events.rst
Comment (mariajgrimaldi) [reply docs/concepts/openedx-events.rst]: Thank you for the suggestion. I added a reference to how to listen to signals in Django: https://github.com/openedx/openedx-events/pull/406/commits/6926b20ca45c80f201dd547123749aea02f3253a#diff-982f9d143e6afcea5b84eeb1ea3c3fb5412239c1968359fab161cb4af58e9b63R63

File: docs/concepts/openedx-events.rst
Comment (mariajgrimaldi) [reply docs/concepts/openedx-events.rst]: Fixed, thank you!

File: docs/concepts/openedx-events.rst
Comment (mariajgrimaldi) [reply docs/concepts/openedx-events.rst]: I updated the step by step with a more accurate description, could you review it again?

File: docs/concepts/openedx-events.rst
Comment (mariajgrimaldi) [reply docs/concepts/openedx-events.rst]: I meant something like:

_The event processing is complete once all signal receivers have finished executing, so the application continues its execution..._

But that sounds way more complex than it actually is, so I changed it: https://github.com/openedx/openedx-events/pull/406/commits/6926b20ca45c80f201dd547123749aea02f3253a#diff-982f9d143e6afcea5b84eeb1ea3c3fb5412239c1968359fab161cb4af58e9b63R37

File: docs/concepts/openedx-events.rst
Comment (mariajgrimaldi) [reply docs/concepts/openedx-events.rst]: Absolutely! I'm working on a list of use cases. In the next few hours, I'll link the PR here in this comment.

File: docs/concepts/openedx-events.rst
Comment (mariajgrimaldi) [reply docs/concepts/openedx-events.rst]: There is no we. I changed it: https://github.com/openedx/openedx-events/pull/406/commits/6926b20ca45c80f201dd547123749aea02f3253a#diff-982f9d143e6afcea5b84eeb1ea3c3fb5412239c1968359fab161cb4af58e9b63R31

File: docs/concepts/openedx-events.rst
Comment (sarina) [reply docs/concepts/openedx-events.rst]: Overall it Looks great now, I'll do a final review once these use cases are in

Comment (mariajgrimaldi) [reply docs/concepts/openedx-events.rst]: I haven't had the time to work on this. So I'll be on it today. Thank you for the patience!

Comment (mariajgrimaldi) [reply docs/concepts/openedx-events.rst]: Here's the PR with the use cases: https://github.com/openedx/openedx-events/pull/416

**Comment by mariajgrimaldi** [comment]
@felipemontoya @Ian2012: Could you quickly verify that everything is technically correct from a framework maintainer's perspective? Thanks!

**Review by sarina — APPROVED** [review]

**Review by Ian2012 — APPROVED** [review]

**Comment by mariajgrimaldi** [comment]
I'll merge this once the CLA check passes. I'm not sure why it's taking so long.

**Pull request merged**
